modules/backend/clinic/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+set(CLINIC_SOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/Clinic.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/room/Room.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/staff/doctor/Doctor.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/staff/Staff.cpp
+)
+
+add_library(clinic-lib STATIC ${CLINIC_SOURCES})
+
+target_compile_options(clinic-lib PRIVATE -Werror -Wall -Wextra -pedantic)
+
+target_link_libraries(clinic-lib PUBLIC
+                                 common-lib
+                                 warehouse-lib
+)
+
+target_include_directories(clinic-lib PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+
+# install created lib in our own libraries directory
+install(TARGETS clinic-lib
+        ARCHIVE DESTINATION ${OWN_LIBRARIES_DIR})
+
+# copy include files to our own libraries directory
+install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/ DESTINATION ${OWN_LIBRARIES_DIR}/include
+        FILES_MATCHING PATTERN "*.h" PATTERN "*.hpp")
```

Comment:
Please clarify me if I am wrong.

1st install: installs clinic library as archive to own_lib_dir
2nd install: copy header files to own_lib_dir/include 

Why is that so? I am trying to understand purpose, pros and cons

Reply:
Yes thats correct.
First install install binary libraries. The second - copies headers.
Headers are necessary, as in targets using libs any include without them would not work. I was surprised too.
This was suggested by copilot.
But i tested it anyway as first the headers were not installed - and none include in tests worked.
In the first approach I had one backend lib. But as I saw the problem with includes, I decided for seperate libs.

All in all if we want one backend lib - I think that it will be possible but install of includes most likely will have to stay.
Unless we find a better solution of course.

It's hard to find pros and cons so far as I see no other solution at this point. Perhaps there are some package managers, which could make something better.
I wonder as well, if we add header files to sources together with .cpp maybe they would also be included in lib it self.